solcx/install.py

```diff
@@ -65,9 +66,33 @@ def get_solc_folder(solcx_binary_path=None):
         return path
 
 
+def get_version_location(version, solcx_binary_path=None):
+    folder_location = get_solc_folder(solcx_binary_path=solcx_binary_path)
+    version_location = folder_location.joinpath(f"solc-v{version}")
+    return version_location
+
+
+def is_version_installed(version_location=None, version=None, solcx_binary_path=None):
+    if version_location is None:
+        version_location = get_version_location(version, solcx_binary_path)
+    if version_location.exists():
+        LOGGER.info(f"solc version already installed at: {version_location}")
+        return True
+    return False
+
+
 def _import_version(path):
-    version = subprocess.check_output([path, "--version"]).decode()
-    return "v" + version[version.index("Version: ") + 9 : version.index("+")]
+    version_str = subprocess.check_output([path, "--version"]).decode()
+    version = Version(version_str[version_str.index("Version: ") + 9 : version_str.index("+")])
+    return version
+
+
+def _check_version(version):
+    if type(version) != "semantic_version.base.Version":
+        version = Version(str(version).replace("v", "").replace(".exe", ""))
```

Comment:
What is the case where stripping `.exe` is necessary?  This method only handles version strings not file paths, as far as I can see.

Reply:
I did this based on `get_executable`:
```python
def get_executable(version=None, solcx_binary_path=None):
    if not version and not solc_version:
        raise SolcNotInstalled(
            f"Solc is not installed. Call solcx.get_available_solc_versions()"
            f" to view for available versions and solcx.install_solc() to install."
        )
    if not version:
        version = solc_version
    solc_bin = get_version_location(version, solcx_binary_path)
    if sys.platform == "win32":
        solc_bin = solc_bin.joinpath("solc.exe")
    if not solc_bin.exists():
        raise SolcNotInstalled(
            f"Solc {version} has not been installed."
            f" Use solcx.install_solc('{version}') to install."
        )
    return solc_bin
```

This function is used several times for different purposes and I was concerned the `.exe` would randomly break something.  I'll remove the replacement and see how it goes :)